review/internal/api/api.go

```diff
@@ -1,81 +1,37 @@
 package api
 
 import (
-	"context"
-	"coupon_service/internal/service/entity"
-	"fmt"
-	"log"
-	"net/http"
-	"time"
-
 	"github.com/gin-gonic/gin"
+
+	"coupon_service/internal/service/entity"
 )
 
 type Service interface {
-	ApplyCoupon(entity.Basket, string) (*entity.Basket, error)
-	CreateCoupon(int, string, int) any
-	GetCoupons([]string) ([]entity.Coupon, error)
-}
-
-type Config struct {
-	Host string
-	Port int
+	ApplyCoupon(int, int, string) (entity.Basket, error)
+	CreateCoupon(int, string, int) error
+	ListCoupons(...string) ([]entity.Coupon, error)
 }
 
 type API struct {
-	srv *http.Server
```

Comment:
I splitted API logic into 2 packages:
API which is responsible for handling endpoints
App which is responsible for starting and running http server.